extension/encoding/googlecloudlogentryencodingextension/internal/vpcflowlog/parser.go

```diff
@@ -0,0 +1,329 @@
+			shared.PutStr(gcpVPCFlowDestInstanceMIGName, inst.ManagedInstanceGroup.Name, attr)
+			shared.PutStr(gcpVPCFlowDestInstanceMIGZone, inst.ManagedInstanceGroup.Zone, attr)
+		}
+	}
+}
+
+func handleLocation(loc *location, side flowSide, attr pcommon.Map) {
+	if loc == nil {
+		return
+	}
+
+	switch side {
+	case src:
+		shared.PutInt(gcpVPCFlowSourceASN, loc.ASN, attr)
+		shared.PutStr(gcpVPCFlowSourceGeoCity, loc.City, attr)
+		shared.PutStr(gcpVPCFlowSourceGeoContinent, loc.Continent, attr)
+		shared.PutStr(gcpVPCFlowSourceGeoCountry, loc.Country, attr)
+		shared.PutStr(gcpVPCFlowSourceGeoRegion, loc.Region, attr)
+	case dest:
+		shared.PutInt(gcpVPCFlowDestASN, loc.ASN, attr)
+		shared.PutStr(gcpVPCFlowDestGeoCity, loc.City, attr)
+		shared.PutStr(gcpVPCFlowDestGeoContinent, loc.Continent, attr)
+		shared.PutStr(gcpVPCFlowDestGeoCountry, loc.Country, attr)
+		shared.PutStr(gcpVPCFlowDestGeoRegion, loc.Region, attr)
+	}
+}
+
+func handleVPC(vpc *vpc, side flowSide, attr pcommon.Map) {
+	if vpc == nil {
+		return
+	}
+
+	switch side {
+	case src:
+		shared.PutStr(gcpVPCFlowSourceProjectID, vpc.ProjectID, attr)
+		shared.PutStr(gcpVPCFlowSourceSubnetName, vpc.SubnetworkName, attr)
+		shared.PutStr(gcpVPCFlowSourceSubnetRegion, vpc.SubnetworkRegion, attr)
+		shared.PutStr(gcpVPCFlowSourceVPCName, vpc.VPCName, attr)
+	case dest:
+		shared.PutStr(gcpVPCFlowDestProjectID, vpc.ProjectID, attr)
+		shared.PutStr(gcpVPCFlowDestSubnetName, vpc.SubnetworkName, attr)
+		shared.PutStr(gcpVPCFlowDestSubnetRegion, vpc.SubnetworkRegion, attr)
+		shared.PutStr(gcpVPCFlowDestVPCName, vpc.VPCName, attr)
+	}
+}
+
+func handleInternetRoutingDetails(ird *internetRoutingDetails, attr pcommon.Map) {
+	if ird == nil || len(ird.EgressASPath) == 0 {
+		return
+	}
+
+	asPaths := attr.PutEmptySlice(gcpVPCFlowEgressASPaths)
+	for _, path := range ird.EgressASPath {
+		pathMap := asPaths.AppendEmpty().SetEmptyMap()
+		asDetails := pathMap.PutEmptySlice("as_details")
+		for _, detail := range path.ASDetails {
+			detailMap := asDetails.AppendEmpty().SetEmptyMap()
+			if detail.ASN != nil {
+				detailMap.PutInt("asn", *detail.ASN)
+			}
+		}
+	}
+}
+
+func ParsePayloadIntoAttributes(payload []byte, attr pcommon.Map) error {
+	var log vpcFlowLog
+	if err := gojson.Unmarshal(payload, &log); err != nil {
+		return fmt.Errorf("failed to unmarshal VPC flow log payload: %w", err)
+	}
+
+	// Handle connection details
+	handleConnection(log.Connection, attr)
+
+	// Handle basic flow information
+	shared.PutStr(gcpVPCFlowReporter, log.Reporter, attr)
+
+	if err := shared.AddStrAsInt(gcpVPCFlowBytesSent, log.BytesSent, attr); err != nil {
+		return fmt.Errorf("failed to add bytes sent: %w", err)
+	}
+
+	if err := shared.AddStrAsInt(gcpVPCFlowPacketsSent, log.PacketsSent, attr); err != nil {
+		return fmt.Errorf("failed to add packets sent: %w", err)
+	}
+
+	shared.PutStr(gcpVPCFlowStartTime, log.StartTime, attr)
+	shared.PutStr(gcpVPCFlowEndTime, log.EndTime, attr)
```

Comment:
Let's convert this to unix nano and add it as an int. To keep consistent that every timestamp has the [same format](https://github.com/open-telemetry/opentelemetry-collector/blob/14a78328c6abbbced62ebd1e273dd200e46cc7fb/pdata/pcommon/timestamp.go#L10-L12).